samples/widgets/sales-dashboard-widgets/RevenueByProductWithUserPreferences/src/RevenueByProductWithUserPreferences.jsx

```diff
@@ -0,0 +1,138 @@
+            ['ALL', 'ALL', 'Middle East', 'IOTServer', 91472.91, 14, 8],
+            ['ALL', 'ALL', 'North America', 'Analytics', 324886.4, 37, 23],
+            ['ALL', 'ALL', 'North America', 'APIManager', 774886.4, 70, 44],
+            ['ALL', 'ALL', 'North America', 'ESB', 1134886.4, 40, 24],
+            ['ALL', 'ALL', 'North America', 'IdentityServer', 390128.28, 30, 12],
+            ['ALL', 'ALL', 'North America', 'IOTServer', 0, 0, 0],
+            ['ALL', 'ALL', 'ROW', 'Analytics', 114522.14, 13, 8],
+            ['ALL', 'ALL', 'ROW', 'APIManager', 214522.14, 19, 10],
+            ['ALL', 'ALL', 'ROW', 'ESB', 174522.14, 15, 7],
+            ['ALL', 'ALL', 'ROW', 'IdentityServer', 224522.14, 25, 17],
+            ['ALL', 'ALL', 'ROW', 'IOTServer', 0, 0, 0]
+        ];
+
+        this.state = {
+            data: this.overallProductData,
+            width: props.glContainer.width,
+            height: props.glContainer.height
+
+        };
+
+        this.metadata = {
+            names: ['Country', 'Country Code', 'Region', 'Product', 'Revenue', 'Downloads', 'Customers'],
+            types: ['ordinal', 'ordinal', 'ordinal', 'ordinal', 'linear', 'linear', 'linear']
+        };
+
+        this.barChartConfig = {
+            x: 'Product',
+            charts: [{type: 'bar', y: 'Revenue', fill: '#10c469'}],
+            maxLength: 5,
+            width: props.glContainer.width,
+            height: props.glContainer.height,
+            animate: true
+        };
+
+        this.setReceivedMsg = this.setReceivedMsg.bind(this);
+        this.handleResize = this.handleResize.bind(this);
+        this.props.glContainer.on('resize', this.handleResize);
+    }
+
+    handleResize() {
+        this.setState({width: this.props.glContainer.width, height: this.props.glContainer.height});
+    }
+
+    componentWillMount() {
+        super.subscribe(this.setReceivedMsg);
+    }
+
+    setReceivedMsg(receivedMsg) {
+        if (receivedMsg.selectedRegion === "ALL") {
+            this.setState({data: this.overallProductData})
+        } else {
+            let array = [];
+            this.rowData.map(dataElement => {
+                if (dataElement[2] === receivedMsg.selectedRegion) {
+                    array.push(dataElement);
+                }
+            });
+            this.setState({data: array});
+        }
+    }
+
+    render() {
+        return (
+            <div
+                style={{ height: this.props.glContainer.height, width: this.props.glContainer.width, paddingBottom: 10 }}
+            >
+                <VizG
+                    config={this.barChartConfig}
+                    metadata={this.metadata}
+                    data={this.state.data}
+                    width={this.props.glContainer.width}
+                    height={this.props.glContainer.height}
+                />
+            </div>
+        );
+    }
+}
+
+global.dashboard.registerWidget("RevenueByProduct", RevenueByProduct);
```

Comment:
Please use single quotes

Reply:
I think it is better to keep it as it is to be inline with the other widgets

Reply:
ESLint specifications enforce to use single quotes for strings. It's always better to follow those